packages/arb-token-bridge-ui/src/components/App/App.tsx

```diff
@@ -90,7 +90,7 @@ const ArbTokenBridgeStoreSyncWrapper = (): JSX.Element | null => {
       selectedTokenAddress === nativeCurrency.address ||
       selectedTokenL2Address === nativeCurrency.address
     ) {
-      actions.app.setSelectedToken(null)
+      actions.app.setSelectedToken({ token: null })
```

Comment:
`setSelectedToken` payload has changed to include an optional flag `isSelectedTokenEther`.

When `isSelectedTokenEther === true` and `token === null`, everything will assume that the native currency is Ether.

When `isSelectedTokenEther === false` and `token === null`, everything will work as previously, which means it will default to the native currency on the child chain. Note this still could be ETH, `isSelectedTokenEther` just serves as an override to cover custom gas token scenarios.

See `actions.ts` and `state.ts` for more.